packages/app-desktop/gui/MainScreen/MainScreen.tsx

```diff
@@ -185,6 +186,15 @@ class MainScreenComponent extends React.Component<Props, State> {
 		this.layoutModeListenerKeyDown = this.layoutModeListenerKeyDown.bind(this);
 
 		window.addEventListener('resize', this.window_resize);
+
+		ipcRenderer.on('asynchronous-message', (_event: any, message: string, args: any) => {
+			if (message === 'openUrl') {
```

Comment:
I'm having a hard time following the code but shouldn't this message be called "openNote"?

Reply:
Ok my confusion comes from the name "parseUrl" which is used to parse regular URLs in the code, but this is your own implementation. I think we need to name things in a more specific way because all this deals with callback URLs.

For example "parseCallbackUrl". And "initialUrl" should probably also be called "initialCallbackUrl".